File: backend/insertionSort.js
function insertionSort(a)
{
    for (let i = 1; i < a.length; i++) {
      let j = i - 1;
      let tmp = a[i];
      while (j >= 0 && a[j] > tmp) {
        a[j + 1] = a[j]
        j--
      }
      a[j+1] = tmp
    }
    return a
}

console.log(insertionSort([9,1,3,7,2]));

File: Paralelización/Estables/insertionSort.c
#include <omp.h>
#include <stdio.h>
#include <stdlib.h>

#define SIZE 1000

/* 
  Function Insertion: this function organizes the data array using Insertion Sort
  @param: int A[], sends an int data array 
  @return: float with the execution time
*/
float insertionSort(int A[])
{
    int N = SIZE;
	int i=0, j=0;
	int first;
	double start,end;
	start= omp_get_wtime(); //
    int  key;  
    for (i = 1; i < SIZE; i++) 
    {  
        {
            key = A[i];  
             j = i - 1;  
             while (j >= 0 && A[j] > key) 
            {  
                A[j + 1] = A[j];  
                j = j - 1;  
            }  
            A[j + 1] = key; 
        }
         
    }  

    end=omp_get_wtime();
    printf("\n-------------------------\n Time Parallel= %f",(end-start));
    return end-start;

}

int main () {
	//int SIZE =1000; //tamaño total
	int A[SIZE];
	for(int i=0;i<SIZE;i++)
	{
	    A[i]=rand()%SIZE;
	}
	insertionSort(A);
}
